lib/os/fdtable.c

```diff
@@ -219,6 +219,12 @@ int ioctl(int fd, unsigned long request, ...)
 	return res;
 }
 
+/*
+ * In the socket offload case, we have yet to add support for the fdtable
+ * feature. The socket offload subsys has already defined fcntl, hence we
+ * avoid redefining fcntl here.
+ */
+#ifndef CONFIG_NET_SOCKETS_OFFLOAD
```

Comment:
The fcntl() is generic function that should be available even if we have defined CONFIG_NET_SOCKETS_OFFLOAD. Now you are preventing the use of it for those file descriptors that are not offloaded.

Reply:
Changed to making it SimpleLink-specific.